### DISCUSSION TITLE
Authentication when running a local instance

### DISCUSSION THREAD

**Discussion by arrabito** [DIRACGrid/diracx#77] [Developer discussions]
Dear all,

I was wondering if when running a local instance of diracx with ./run_local.sh one should be able to authenticate.

I've run the ./run_local.sh script and then use the wen dex interface at:

http://127.0.0.1:8000/doc

However when I click on Authorize and I then select:

vo:diracAdmin

I get:

{"detail":"Unrecognised redirect_uri"}

Is it suppose to work?

Thanks.

P.S. I'm on MacOS

**Comment by chaen** [comment]
the `run_local.sh` script does not start a `dex` instance. You need a running IdP for that (dex, IAM, etc)

**Comment by arrabito** [comment]
I see. So how is intended to be used?
Is it intended to be used just as a demonstrator to make sure that services start correctly?
Or it assumes that one runs an IdP separately?
Thanks for the clarifications.

**Comment by chrisburr** [comment]
I think we should probably remove the `run_local.sh` script entirely as I'm not sure there is really a situation in which it is usable in a useful way.

**Comment by arrabito** [comment]
If there is no useful usage, then I agree since it can be confusing for the new comers.
By the way, is there already a way in diracx to use/configure a given IdP? For instance in CTA we have an IaM instance that I would like to try.
I'm asking this here, just in case it can be associated with the run_local.sh script.

**Comment by chaen** [comment]
I think the `run_local.sh` is very useful exactly for that sort of situation. 
If I want to test something against the LHCb certification instance, I find it much easier and faster to configure just the IAM instance and the DB connection string in this bash script, than playing with `helm upgrade`. 
And I think that holds true for a lot of people
